File: browser/hns_ads/android/java/org/chromium/chrome/browser/HnsAdsNativeHelper.java
/**
 * Copyright (c) 2019 The Hns Authors. All rights reserved.
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at https://mozilla.org/MPL/2.0/.
 */

package org.chromium.chrome.browser;

import org.chromium.base.annotations.JNINamespace;
import org.chromium.base.annotations.NativeMethods;
import org.chromium.chrome.browser.profiles.Profile;

@JNINamespace("hns_ads")
public class HnsAdsNativeHelper {
    private HnsAdsNativeHelper() {}

    public static boolean nativeIsOptedInToNotificationAds(Profile profile) {
        return HnsAdsNativeHelperJni.get().isOptedInToNotificationAds(profile);
    };

    public static void nativeSetOptedInToNotificationAds(Profile profile, boolean opted_in) {
        HnsAdsNativeHelperJni.get().setOptedInToNotificationAds(profile, opted_in);
    };

    public static boolean nativeIsSupportedRegion(Profile profile) {
        return HnsAdsNativeHelperJni.get().isSupportedRegion(profile);
    };

    public static void nativeOnNotificationAdShown(Profile profile, String j_notification_id) {
        HnsAdsNativeHelperJni.get().onNotificationAdShown(profile, j_notification_id);
    };

    public static void nativeOnNotificationAdClosed(
            Profile profile, String j_notification_id, boolean j_by_user) {
        HnsAdsNativeHelperJni.get().onNotificationAdClosed(profile, j_notification_id, j_by_user);
    };

    public static void nativeOnNotificationAdClicked(Profile profile, String j_notification_id) {
        HnsAdsNativeHelperJni.get().onNotificationAdClicked(profile, j_notification_id);
    };

    @NativeMethods
    interface Natives {
        boolean isOptedInToNotificationAds(Profile profile);
        void setOptedInToNotificationAds(Profile profile, boolean opted_in);
        boolean isSupportedRegion(Profile profile);
        void onNotificationAdShown(Profile profile, String j_notification_id);
        void onNotificationAdClosed(Profile profile, String j_notification_id, boolean j_by_user);
        void onNotificationAdClicked(Profile profile, String j_notification_id);
    }
}

File: browser/hns_ads/android/hns_ads_native_helper.cc
/* Copyright (c) 2019 The Hns Authors. All rights reserved.
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at https://mozilla.org/MPL/2.0/. */

#include "hns/browser/hns_ads/android/hns_ads_native_helper.h"

#include <string>

#include "base/android/jni_string.h"
#include "hns/browser/hns_ads/ads_service_factory.h"
#include "hns/browser/hns_ads/android/jni_headers/HnsAdsNativeHelper_jni.h"
#include "hns/components/hns_ads/core/public/ads_util.h"
#include "hns/components/hns_ads/core/public/prefs/pref_names.h"
#include "chrome/browser/profiles/profile_android.h"
#include "components/prefs/pref_service.h"

namespace hns_ads {

// static
jboolean JNI_HnsAdsNativeHelper_IsOptedInToNotificationAds(
    JNIEnv* env,
    const base::android::JavaParamRef<jobject>& j_profile_android) {
  Profile* profile = ProfileAndroid::FromProfileAndroid(j_profile_android);
  return profile->GetPrefs()->GetBoolean(
      hns_ads::prefs::kOptedInToNotificationAds);
}

// static
void JNI_HnsAdsNativeHelper_SetOptedInToNotificationAds(
    JNIEnv* env,
    const base::android::JavaParamRef<jobject>& j_profile_android,
    jboolean should_enable_ads) {
  Profile* profile = ProfileAndroid::FromProfileAndroid(j_profile_android);
  profile->GetPrefs()->SetBoolean(hns_ads::prefs::kOptedInToNotificationAds,
                                  should_enable_ads);
}

// static
jboolean JNI_HnsAdsNativeHelper_IsSupportedRegion(
    JNIEnv* env,
    const base::android::JavaParamRef<jobject>& j_profile_android) {
  Profile* profile = ProfileAndroid::FromProfileAndroid(j_profile_android);
  AdsService* ads_service = AdsServiceFactory::GetForProfile(profile);
  if (!ads_service) {
    return false;
  }

  return hns_ads::IsSupportedRegion();
}

// static
void JNI_HnsAdsNativeHelper_OnNotificationAdShown(
    JNIEnv* env,
    const base::android::JavaParamRef<jobject>& j_profile_android,
    const base::android::JavaParamRef<jstring>& j_notification_id) {
  Profile* profile = ProfileAndroid::FromProfileAndroid(j_profile_android);
  AdsService* ads_service = AdsServiceFactory::GetForProfile(profile);
  if (!ads_service) {
    return;
  }

  const std::string notification_id =
      base::android::ConvertJavaStringToUTF8(env, j_notification_id);
  ads_service->OnNotificationAdShown(notification_id);
}

// static
void JNI_HnsAdsNativeHelper_OnNotificationAdClosed(
    JNIEnv* env,
    const base::android::JavaParamRef<jobject>& j_profile_android,
    const base::android::JavaParamRef<jstring>& j_notification_id,
    jboolean j_by_user) {
  Profile* profile = ProfileAndroid::FromProfileAndroid(j_profile_android);
  AdsService* ads_service = AdsServiceFactory::GetForProfile(profile);
  if (!ads_service) {
    return;
  }

  const std::string notification_id =
      base::android::ConvertJavaStringToUTF8(env, j_notification_id);
  ads_service->OnNotificationAdClosed(notification_id, j_by_user);
}

// static
void JNI_HnsAdsNativeHelper_OnNotificationAdClicked(
    JNIEnv* env,
    const base::android::JavaParamRef<jobject>& j_profile_android,
    const base::android::JavaParamRef<jstring>& j_notification_id) {
  Profile* profile = ProfileAndroid::FromProfileAndroid(j_profile_android);
  AdsService* ads_service = AdsServiceFactory::GetForProfile(profile);
  if (!ads_service) {
    return;
  }

  const std::string notification_id =
      base::android::ConvertJavaStringToUTF8(env, j_notification_id);
  ads_service->OnNotificationAdClicked(notification_id);
}

}  // namespace hns_ads
